File: Day-02/java/LuftigerLuca/Day2.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

public class Day2 {

    public static List<String> movement = new ArrayList<>();

    public static void main(String[] args) {
        getMovement();

        //Part1
        int positionForward = 0;
        int positionDepth = 0;

        for (String move : movement) {
            String[] splitMove = move.split(" ");
            switch (splitMove[0].toLowerCase()){
                case "forward" -> positionForward += Integer.parseInt(splitMove[1]);
                case "down" -> positionDepth += Integer.parseInt(splitMove[1]);
                case "up" -> positionDepth -= Integer.parseInt(splitMove[1]);
            }
        }

        System.out.println("The submarine is at position " + positionForward * positionDepth);

        //Part2
        int aim = 0;
        positionForward = 0;
        positionDepth = 0;
        for (String move : movement){
            String[] splitMove = move.split(" ");
            int i = Integer.parseInt(splitMove[1]);

            if(splitMove[0].equalsIgnoreCase("down")) aim += i;
            if (splitMove[0].equalsIgnoreCase("up")) aim -= i;
            if(splitMove[0].equalsIgnoreCase("forward")){
              positionForward += i;
              positionDepth += i * aim;
            }
        }

        System.out.println("The submarine is at position " + positionForward * positionDepth);
    }

    public static void getMovement(){
        try {
            movement.addAll(Files.readAllLines(Path.of("src/main/resources/movement.txt")));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Day-02/kotlin/Skyslycer/Day2.kt
import java.nio.file.Files
import java.nio.file.Paths

fun main() {
    val instructions = Files.readAllLines(Paths.get("./inputs/day2/input.txt"))

    var position = 0
    var depth = 0

    // First part
    instructions.map { it.split(" ") }.forEach {
        when (it[0]) {
            "forward" -> position += it[1].toInt()
            "up" -> depth -= it[1].toInt()
            "down" -> depth += it[1].toInt()
        }
    }

    println("First solution: ${position * depth}")

    // Second part
    var aim = 0
    position = 0
    depth = 0

    instructions.map { it.split(" ") }.forEach {
        when (it[0]) {
            "forward" -> {
                position += it[1].toInt()
                depth += aim * it[1].toInt()
            }
            "up" -> aim -= it[1].toInt()
            "down" -> aim += it[1].toInt()
        }
    }

    println("Second solution: ${position * depth}")
}